ocean_lib/ocean/ocean_assets.py

```diff
@@ -55,7 +56,13 @@ class OceanAssets:
 
     """Ocean assets class."""
 
-    def __init__(self, config, web3, data_provider, ddo_registry_address):
+    def __init__(
+        self,
+        config: Config,
+        web3: Web3,
+        data_provider: object,
```

Comment:
`DataServiceProvider`

Reply:
This one is compromised by the DataProviderMock. Inside the tests we use DataProviderMock objects, instead of DataServiceProvider. Should we add both as an option? I don't particularly like the usage of a Mock type hint inside the main code.

Reply:
I think that `DataProviderMock` should be a subclass of `DataProviderService` instead of `object`.  This would allows the main code to enforce the `DataProviderService` type.

Reply:
Fixed.